src/malli/json_schema.cljc

```diff
@@ -7,12 +7,21 @@
 (defprotocol JsonSchema
   (-accept [this children options] "transforms schema to JSON Schema"))
 
-(defn -ref [x] {:$ref (str "#/definitions/" x)})
+(defn -ref [x] {:$ref (apply str "#/definitions/"
+                             (cond
+                               (qualified-keyword? x) [(namespace x) "~1"
+                                                       (name x)]
+                               (keyword? x) [(name x)]
+                               :else [x]))})
 
 (defn -schema [schema {::keys [transform definitions] :as options}]
   (let [result (transform (m/deref schema) options)]
     (if-let [ref (m/-ref schema)]
-      (do (swap! definitions assoc ref result) (-ref ref))
+      (do
+        (when-not (and (contains? result :$ref)
+                       (= (:$ref result) ref)) ; don't create circular defs
+          (swap! definitions assoc ref result))
+        (-ref ref))
       result)))
```

Comment:
I'd like to see a test for this new behaviour.

Reply:
Hmm... I'm not sure what causes it. I just noticed that both the actual definition and a circular ref were being generated in here from my schema.

I'll see if I can figure out a minimal repro for a test.

Reply:
I just bumped into this:
```
user=> (malli.json-schema/transform (mu/closed-schema [:schema {:registry {"Foo" :int}} "Foo"]))
{:$ref "#/definitions/Foo", :definitions {"Foo" {:$ref "#/definitions/Foo"}}}
user=> (malli.json-schema/transform [:schema {:registry {"Foo" :int}} "Foo"])
{:$ref "#/definitions/Foo", :definitions {"Foo" {:type "integer"}}}
```
Your PR seems to fix it. I created issue #868 for this bug.

Reply:
Turned that into a test in 64b15cb40a73ca40b22a9647c85ddc601e8b91e0.